src/bin/pg_autoctl/supervisor.c

```diff
@@ -256,6 +567,15 @@ supervisor_loop(Supervisor *supervisor)
 			{
 				Service *dead = NULL;
 
+				/* if this is a dynamic service which just exited, do nothing */
+				if (supervisor_find_recently_removed_dynamic(supervisor,
```

Comment:
As mentioned already, I think we need a better name for the structure slot and the function here.

Reply:
Absolutely. Will use your suggestions.